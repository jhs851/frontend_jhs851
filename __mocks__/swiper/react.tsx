import React from 'react'

const mock: React.FC = ({ children }) => <div>{children}</div>

export default mock
