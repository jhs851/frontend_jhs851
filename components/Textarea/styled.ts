import styled from 'styled-components'
import { hideScrollbar } from 'styles/scrollbar'

export const StyledTextarea = styled.textarea`
  font-family: AppleSDGothicNeo, sans-serif;
  font-size: 14px;
  letter-spacing: -0.12px;
  color: ${({ theme }) => theme.foreground};
  background-color: ${({ theme }) => theme.background};
  width: 100%;
  resize: none;
  ${hideScrollbar}

  ::placeholder {
    color: ${({ theme }) => theme.battleshipGrey};
  }
`
