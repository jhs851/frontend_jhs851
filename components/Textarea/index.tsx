import React, { TextareaHTMLAttributes } from 'react'
import { StyledTextarea } from './styled'

type P = TextareaHTMLAttributes<Element> & {
  className?: string
}

const Textarea: React.ForwardRefRenderFunction<HTMLTextAreaElement, P> = (
  props,
  ref,
) => <StyledTextarea ref={ref} {...props} />

export default React.forwardRef(Textarea)
