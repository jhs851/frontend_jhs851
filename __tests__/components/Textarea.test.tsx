import { render } from '@testing-library/react'
import Textarea from 'components/Textarea'

describe('Textarea 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    expect(render(<Textarea />).container).toMatchSnapshot()
  })
})
