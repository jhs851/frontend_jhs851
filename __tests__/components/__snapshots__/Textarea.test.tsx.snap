// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea 컴포넌트 렌더링이 올바르게 된다 1`] = `
.c0 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0::-webkit-scrollbar {
  display: none;
}

<div>
  <textarea
    class="c0"
  />
</div>
`;
